api/src/main/java/org/openmrs/module/appointments/AppointmentsActivator.java

```diff
@@ -27,18 +30,34 @@ public class AppointmentsActivator extends BaseModuleActivator {
 	private Log log = LogFactory.getLog(this.getClass());
 
```

Comment:
I moved away from the original approach of using ApplicationContextAware, in favor of a more generally understood approach of using the module activator and iterating over registered components.  I made this generalized as it is not specific to advice, but could be utilized for other reasons if needed.

One of the main reasons I moved to this approach is to make an attempt to also clean up / remove advice if the module is stopped like is done with standard advice in the ModuleFactory.  This is just an additional precaution, though OpenMRS doesn't really function properly anyway (or I wouldn't trust it to) if modules are started/stopped via the legacy admin interface, and this isn't really a huge concern.